src/main/scala/edu/ie3/simona/service/primary/PrimaryServiceWorker.scala

```diff
@@ -131,32 +132,48 @@ final case class PrimaryServiceWorker[V <: Value](
             s"Provided init data '${unsupported.getClass.getSimpleName}' for primary service are invalid!"
           )
         )
-    }).map { case (source, simulationStart) =>
+    }).flatMap { case (source, simulationStart) =>
       implicit val startDateTime: ZonedDateTime = simulationStart
 
       val (maybeNextTick, furtherActivationTicks) = SortedDistinctSeq(
         // Note: The whole data set is used here, which might be inefficient depending on the source implementation.
         source.getTimeSeries.getEntries.asScala
-          .filter { timeBasedValue =>
-            val dateTime = timeBasedValue.getTime
-            dateTime.isEqual(simulationStart) || dateTime.isAfter(
-              simulationStart
-            )
-          }
           .map(timeBasedValue => timeBasedValue.getTime.toTick)
+          .filter(_ >= 0L)
           .toSeq
           .sorted
       ).pop
 
-      /* Set up the state data and determine the next activation tick. */
-      val initializedStateData =
-        PrimaryServiceInitializedStateData(
-          maybeNextTick,
-          furtherActivationTicks,
-          simulationStart,
-          source,
-        )
-      (initializedStateData, maybeNextTick)
+      (maybeNextTick, furtherActivationTicks) match {
+        case (maybeNextTick @ Some(tick), furtherActivationTicks)
```

Comment:
You're shadowing the `maybeNextTick` variable unnecessarily here

```suggestion
        case (Some(tick), furtherActivationTicks)
```

Reply:
Just committed this myself because it's my only suggestion here...